merkletree/proof.go

```diff
@@ -59,7 +76,9 @@ func (ap *AuthenticationPath) authPathHash() []byte {
 	return hash
 }
 
-// Verify should be called after the vrf index is verified successfully
+// Verify verifies the authentication path against
```

Comment:
"Verify verifies" sounds awkward. I would be a bit more specific about what this function does: "Verify recomputes the tree's root node from the authentication path, and compares it to the tree hash in the corresponding signed tree root".